skrf/network.py

```diff
@@ -2014,11 +2015,11 @@ def copy(self) -> Network:
             Copy of the Network
 
         """
-        ntwk = Network(s=self.s,
-                       frequency=self.frequency,
-                       z0=self.z0, s_def=self.s_def,
-                       comments=self.comments
-                       )
+        ntwk = Network(z0=self.z0, s_def=self.s_def, comments=self.comments)
+
+        ntwk._s = self.s.copy()
+        ntwk.frequency = self.frequency
```

Comment:
I think `ntwk.frequency` is not immutable yet, however I think it was decided once, that it should be immutable.
But for now it's probably best to call `self.frequency.copy()`

Reply:
Yes, I completely agree that `ntwk.frequency` should remain immutable.

Before submitting this PR, I verified whether `ntwk.frequency` is just a shallow copy or a view of `self.frequency`. It turns out to be safe: `ntwk.frequency` has a different ID than `self.frequency`, and modifying `self.frequency` does not affect the copied `ntwk.frequency`.

```python
>>> import skrf as rf
>>> import numpy as np
>>> 
>>> frequency = rf.Frequency(1,10,101,'ghz')
>>> s = np.random.rand(101, 2, 2) + np.random.rand(101, 2, 2) * 1.j
>>> 
>>> ntwk = rf.Network(s=s, frequency=frequency)
>>> ntwk2 = ntwk.copy()
>>> 
>>> # Check frequency id 
>>> id(ntwk.frequency)
138540906669776
>>> id(ntwk2.frequency)
138540906742544
>>> # Check frequency points
>>> ntwk.frequency == ntwk2.frequency
True
>>> # Check frequency attribute independence/immutability
>>> ntwk.frequency = rf.Frequency.from_f(ntwk.frequency.f[:50])
>>> ntwk.frequency
1000000000.0-5410000000.0 Hz, 50 pts
>>> ntwk2.frequency
1.0-10.0 GHz, 101 pts
```

I think the `new_frequency.copy()` operation automatically performed in the `@frequency.setter` method ensures the immutability of `frequency` even when `self.frequency` is used instead of `self.frequency.copy()`.

https://github.com/scikit-rf/scikit-rf/blob/3ba8bf84aa07f9dab0ab70992c70dbc178f7929e/skrf/network.py#L1372-L1383

Therefore, I think calling `self.frequency` directly can provide higher performance while maintaining the safety and integrity of the result.

I look forward to hearing your thoughts on this and discussing it further.

Reply:
great